### DISCUSSION TITLE
Check auf doppelte Dokumente

### DISCUSSION THREAD

**Discussion by slohrum** [OPUS4/community#7] [Ideas]
Über SWORD werden Dokumente u.U. mehrfach geliefert. 
Zum einen können via DeepGreen neuere Versionen übermittelt werden (einzelne Verlage liefern Vorabversionen), zum anderen teilen sich mehrere Institutionen ein (Fach-)Repositorium.
Doppelte Dokumente können anhand der DOI (alle via DeepGreen gelieferten Dokumente enthalten eine DOI).
Hier wäre eine einfache Prüfmöglichkeit im SWORD-Ingest Bereich nützlich.

**Comment by j3nsch** [comment]
Gibt es eine Möglichkeit zwischen den Institutionen zu unterscheiden, also für wenn die Dokumente gerade übermittelt werden? Damit könnte man die Dokumente dann automatisch zu unterschiedlichen Sammlungen hinzufügen, was vielleicht über die doppelten Dokumente hinaus in dieser Situation nützlich wäre. Eine Möglichkeit zur Unterscheidung wären separate OPUS-Accounts für jede Institution. Es gibt die Funktionalität noch nicht, aber es wäre relativ einfach in Abhängigkeit vom Account weitere Verarbeitungsregel zu definieren, wie z.B. die Zuordnung zu einer Sammlung.

Ansonsten wäre ein "Check für doppelte Dokument" ja generell etwas nützliches. Wäre es in Ordnung das Thema hier auf die konkrete Situation "DeepGreen mit mehrere Institutionen in einer OPUS-Instanz" zu beschränken und den Titel entsprechend anzupassen? "Check auf doppelte Dokument" ist nicht das Problem, um das es geht, sondern eine mögliche Lösung. Um die Handhabbarkeit in der beschriebenen Situation zu verbessern, gibt es aber verschiedene Möglichkeiten.

**Comment by j3nsch** [comment]
@CAWinter Wären separate SWORD-Accounts für die Institutionen machbar oder gibt es noch eine andere Möglichkeit zwischen Dokumenten für die einzelnen Institute zu unterscheiden? 

Der Request ist ja ein POST. Die URL also um GET-Parameter zu erweitern, macht vielleicht Probleme bei der Verarbeitung, könnte aber evtl. auch funktionieren. Eine andere Möglichkeit wäre "Aliases" für die SWORD-URL zu definieren, anhand denen dann zwischen Institutionen unterschieden werden kann. Das ist im Prinzip das gleiche wie ein GET-Parameter, nur das die URI als Parameter interpretiert wird. Das wären Lösungen bei denen DeepGreen nicht erweitert werden muss, um zusätzliche Informationen im SWORD-Protokol oder in HTTP-Headern mit zu übermitteln.

**Reply by CAWinter** [reply]
Separate Sword-Accounts sind machbar für verschiedene Institutionen. Anhand des Enrichment-Felds 
opus.import.user kann man dann identifizieren, wem das Dokument zuzuordnen ist und dieses dann mit einer nachfolgenden Operation einer speziellen Sammlung zuordnen.

Aktuell ist es so, dass man nur eine Import-Sammlung für Sword konfigurieren kann, welcher alle importierten Dokumente zugeordnet werden. 

Dubletten-Checks werden per Sword nicht gemacht. Es ist möglich Dokumente mit der gleichen Doi zu importieren.
